mipengine/node/monetdb_interface/views.py

```diff
@@ -51,21 +51,29 @@ def create_view(
 
     db_execute(view_creation_query)
 
-    if check_min_rows:
-        view_rows_query_result = db_execute_and_fetchall(
-            f"""
-            SELECT COUNT(*)
-            FROM {view_name}
-            """
+    view_rows_query_result = db_execute_and_fetchall(
+        f"""
+        SELECT COUNT(*)
+        FROM {view_name}
+        """
+    )
+    view_rows_result_row = view_rows_query_result[0]
+    view_rows_count = view_rows_result_row[0]
+
+    if view_rows_count < 1:
+        db_execute(f"""DROP VIEW {view_name}""")
+        raise InsufficientDataError(
+            f"{view_rows_count=} "
+            f"Query: {view_creation_query} creates an empty view(zero row count). "
+            f"({view_name=} has been DROPed)"
+        )
+    elif check_min_rows and view_rows_count < MINIMUM_ROW_COUNT:
+        db_execute(f"""DROP VIEW {view_name}""")
+        raise InsufficientDataError(
```

Comment:
I am not sure if we should expose to the controller how many rows are there. Please ask Jason.

Reply:
you re right, I removed the `view_rows_count` from the exception message.